### ISSUE TITLE
[BUG] SCC extraction from mov file results in an empty SCC file

### ISSUE THREAD

**Issue by shazbot666** [CCExtractor/ccextractor#1423]
CCExtractor version:0.94

- regression?{no}
- What platform did you use? {Mac 12.2.1}
- What were the used arguments? `ccextractor -in=mp4 -out=scc -nofc -dru /Volumes/Drive5/Whacked/WhackedOutVideos_short.mov -o /Volumes/Drive5/Whacked/WhackedOutVideos_short.scc'

# Video links
http://videopress.s3.amazonaws.com/samples/Whacked/WhackedOutVideos_short.mov

# Additional information
A .scc file is created, but it has no captions. Captions are displayed in the terminal window in SRT format while the job is running. Tried with multiple mov files. Terminal output here:

ccextractor -in=mp4 -out=scc -nofc -dru /Volumes/Drive5/Whacked/WhackedOutVideos_short.mov -o /Volumes/Drive5/Whacked/WhackedOutVideos_short.scc
CCExtractor 0.94, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: /Volumes/Drive5/Whacked/WhackedOutVideos_short.mov
[Extract: 1] [Stream mode: MP4]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[CEA-708: 63 decoders active]
[CEA-708: using charset "none" for all services]
[Timing mode: Auto] [Debug: No] [Buffer input: No]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .scc] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: No] [Add font typesetting: Yes]
[Convert case: No][Filter profanity: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]
[Quantisation-mode: CCExtractor's internal function]

-----------------------------------------------------------------
Opening file: /Volumes/Drive5/Whacked/WhackedOutVideos_short.mov
Analyzing data with GPAC (MP4 library)
Opening '/Volumes/Drive5/Whacked/WhackedOutVideos_short.mov': ok
Track 1, type=soun subtype=in24
Track 2, type=soun subtype=in24
Track 3, type=soun subtype=in24
Track 4, type=soun subtype=in24
Track 5, type=tmcd subtype=tmcd
Track 6, type=vide subtype=avc1
Track 7, type=clcp subtype=c708
MP4: found 7 tracks: 1 avc and 1 cc
Processing track 1, type=soun subtype=in24
Processing track 2, type=soun subtype=in24
Processing track 3, type=soun subtype=in24
Processing track 4, type=soun subtype=in24
Processing track 5, type=tmcd subtype=tmcd
Processing track 6, type=vide subtype=avc1
Processing track 7, type=clcp subtype=c708
00:00:02:501 --> 00:00:02:501
00:00:02:502 --> 00:00:03:636
Or is this synchronized
00:00:03:637 --> 00:00:05:971
Or is this synchronized
stick poking?
00:00:05:972 --> 00:00:07:173
Or is this synchronized
stick poking?
Actually, those are the rods
00:00:07:174 --> 00:00:08:841
stick poking?
Actually, those are the rods
they use to locate lost people
00:00:08:842 --> 00:00:10:443
Actually, those are the rods
they use to locate lost people
or objects in up to 12 feet
00:00:10:444 --> 00:00:12:478
they use to locate lost people
or objects in up to 12 feet
of powder.
00:00:12:479 --> 00:00:13:512
or objects in up to 12 feet
of powder.
In this case, they find this
00:00:13:513 --> 00:00:15:981
of powder.
In this case, they find this
dude's video camera.
00:00:15:982 --> 00:00:17:350
In this case, they find this
dude's video camera.
And the swear blizzard
00:00:17:351 --> 00:00:18:784
dude's video camera.
And the swear blizzard
starts again.
00:00:18:785 --> 00:00:22:755
And the swear blizzard
starts again.
(bleeping)
00:00:22:756 --> 00:00:24:623
starts again.
(bleeping)
Two-thirds of the world
00:00:24:624 --> 00:00:26:425
(bleeping)
Two-thirds of the world
is covered by water
00:00:28:661 --> 00:00:28:661
Two-thirds of the world
is covered by water
and the rest is covered by us.
00:00:28:662 --> 00:00:30:029
is covered by water
and the rest is covered by us.
Time to gather up all the goofy
100%  |  00:29
Closing media: ok
Found 1 AVC track(s). Found 1 CC track(s).

Total frames time:	  00:00:00:000  (0 frames at 29.97fps)

Min PTS:				00:00:00:000
Max PTS:				00:00:29:996
Length:				 00:00:29:996
Done, processing time = 0 seconds
00:00:30:030 --> 00:00:30:029
and the rest is covered by us.
Time to gather up all the goofy
goings-on from around the globe

No captions were found in input.
Issues? Open a ticket here

**Comment by shazbot666** [comment]
A little more information:
two output files were created. One looks like a temp file, one is the specified -o file. The temp is zero K, the specified output just has the scc header text and nothing else "Scenarist_SCC V1.0"
WhackedOutVideos_short.p0.svc01.scc
WhackedOutVideos_short.scc

**Comment by shazbot666** [comment]
Downgrade from ffmpeg 5 to 4.4. Reinstalled ccextractor 0.94. Same issue.

**Comment by PunitLodha** [comment]
>http://videopress.s3.amazonaws.com/samples/Whacked/WhackedOutVideos_short.mov

Link doesn't work. Please share the video again

**Comment by shazbot666** [comment]
Sorry about that. Try this:
https://drive.google.com/file/d/13p6HBxGXlm0BGpaS15JwCJjfnBdm_Qbm/view?usp=sharing

or if you get a bad MIME type in firefox, try this:
https://drive.google.com/file/d/1lu5jaRs5eY7clTXaB2AFPkYX7qkOh01K/view?usp=sharing

**Comment by PunitLodha** [comment]
>Sorry about that. Try this:
https://drive.google.com/file/d/13p6HBxGXlm0BGpaS15JwCJjfnBdm_Qbm/view?usp=sharing
or if you get a bad MIME type in firefox, try this:
https://drive.google.com/file/d/1lu5jaRs5eY7clTXaB2AFPkYX7qkOh01K/view?usp=sharing

@canihavesomecoffee if you could upload this to SP

**Comment by PunitLodha** [comment]
@shazbot666 Subtitles are being extracted, just not being written to the file. The issue here is that 708 decoder doesn't support SCC format. But, I am not sure if it can support it as SCC seems to be closely linked to 608.

@cfsmp3 any idea if it can be done?

**Comment by cfsmp3** [comment]
> @shazbot666 Subtitles are being extracted, just not being written to the file. The issue here is that 708 decoder doesn't support SCC format. I am not sure if it can support it as SCC seems to be closely linked to 608.
> 
> @cfsmp3 any idea if it can be done?

https://www.rev.com/blog/cea-608-captions-vs-cea-708-captions

```
What file format should I use for CEA-708 captions?

Scenarist (.scc) is the standard caption file format to deliver the CEA-608 and CEA-708 captions.
```

For now (i.e. until that's implemented) if SCC is selected as output we should export 608 instead of 708. Or report an error and make the user be explicit and select 608 :-)

**Comment by shazbot666** [comment]
> For now (i.e. until that's implemented) if SCC is selected as output we should export 608 instead of 708. Or report an error and make the user be explicit and select 608 :-)

That sounds right, I don't think there is any actual 708 data in the test file, its 608 data in a 708 container. If that makes sense. 

From mediainfo:
`Text #1
ID                                       : 7-CC1
Format                                   : EIA-608
Muxing mode                              : Final Cut / CDP
Codec ID                                 : c708`

**Comment by PunitLodha** [comment]
>Or report an error and make the user be explicit and select 608 :-)

We do report an error, but its in the 708 logs, which have to be enabled by using the -708 flag. 
Also, selecting 608 won't work here, because its 608 data in 708 container, so only the 708 decoder can be used here

I'll create a issue for adding SCC support to 708

**Comment by utkarshraj779** [comment]
[CEA-708: 63 decoders active] in the line where this -in=mp4: Input file formet is MP4-out=ssc: Output file format is SSC-nofc: Do not use frame timestamps in this file saver

**Comment by supuram** [comment]
I am reporting here the output I got with the method described here. I was able to replicate the output on the screen but the behaviour of the files were different when i entered them using nano. 

In WSL Debian inside /ccextractor/linux when i write ./ccextractor -out=scc -nofc -dru /mnt/c/users/supra/ClonedProjects/CCExtractor/sccsupport.mov -o sccsupport.scc i get the subtitles on the screen but the last line reads 'No captions were found in input' and i get two output files -  sccsupport.scc which is empty and sccsupport.p0.svc01.scc which is also empty.

When I write ./ccextractor -out=scc -nofc -dru /mnt/c/users/supra/ClonedProjects/CCExtractor/sccsupport.mov -o sccsupport.srt i again see the subtitles on the screen but the last line reads 'No captions were found in input' and i get two output files -  sccsupport.srt which is empty and sccsupport.p0.svc01.scc which is also empty.

 When I write ./ccextractor -out=srt -nofc -dru /mnt/c/users/supra/ClonedProjects/CCExtractor/sccsupport.mov -o sccsupport.srt i again see the subtitles on the screen but the last line reads 'No captions were found in input' and i get two output files -  sccsupport.srt which is empty and sccsupport.p0.svc01.srt which contains the subtitles.
